api/v0/serializers.py

```diff
@@ -1,23 +1,84 @@
 import json
 
+# FIXME: serializers.ValidationError not exceptions.ValidationError
 from django.core import exceptions
 from rest_framework import serializers
 
-from apps.plea.models import Case, UsageStats, Offence, CaseOffenceFilter
+from apps.plea.models import (
+    AuditEvent,
+    Case,
+    CaseOffenceFilter,
+    Offence,
+    UsageStats,
+)
 from apps.result.models import Result, ResultOffence, ResultOffenceData
 from apps.plea.standardisers import standardise_urn
 from apps.plea.validators import is_valid_urn_format
 
 
+class AuditedValidationError(serializers.ValidationError):
+    """A subclass of ValidationErrors that also creates audit events"""
+
+    def __init__(self, *args, **kwargs):
+        event_data = kwargs.pop("event_data")
+        super(AuditedValidationError, self).__init__(*args, **kwargs)
+        AuditEvent().populate(**event_data)
+
+
 class OffenceSerializer(serializers.ModelSerializer):
     class Meta:
         model = Offence
         exclude = ("case",)
 
 
+class AuditEventSerializer(serializers.ModelSerializer):
+    """"""
+
+    class Meta:
+        model = AuditEvent
+        exclude = ('case', 'event_data')
+        required_fields = ("event_type", "event_subtype")
+
+    #def validate_event_type(self, data):
+    #    if data is None:
+    #        raise serializers.ValidationError(
+    #            "Missing event_type creating audit event via API")
+    #    if data == "":
+    #        raise serializers.ValidationError(
+    #            "Parameter, event_type, may not be blank.")
+    #    return data
+
+    #def validate_event_subtype(self, data):
+    #    if data is None:
+    #        raise serializers.ValidationError(
+    #            "Missing event_subtype creating audit event via API")
+    #    if data == "":
+    #        raise serializers.ValidationError(
+    #            "Parameter, event_subtype, may not be blank.")
+    #    return data
+
+    #def validate_event_data(self, data):
+    #    if data is not None:
+    #        try:
+    #            data = json.loads(data)
+    #        except json.ParseError:
+    #            raise serializers.ValidationError(
+    #                "Unable to parse event_data as json")
+    #    else:
+    #        data = {}
+    #    return data
```

Comment:
Yes, now that it's picking up the model's fields I removed allowing null / blank Trues and the validation is inherited correctly.